tensorflow/examples/udacity/1_notmnist.ipynb

```diff
@@ -496,22 +502,61 @@
+        "        dataset, labels = None, None\n",
+        "    return dataset, labels\n",
+        "\n",
+        "def merge_datasets(pickle_files, train_size, valid_size=0):\n",
+        "    num_classes = len(pickle_files)\n",
+        "    valid_dataset, valid_labels = make_arrays(valid_size, image_size)\n",
+        "    train_dataset, train_labels = make_arrays(train_size, image_size)\n",
+        "    vsize_per_class = int(valid_size / num_classes)\n",
+        "    tsize_per_class = int(train_size / num_classes)\n",
+        "    \n",
+        "    start_v, start_t = 0, 0\n",
+        "    end_v, end_t = vsize_per_class, tsize_per_class\n",
+        "    for label, pickle_file in enumerate(pickle_files):       \n",
+        "        try:\n",
+        "            with open(pickle_file, 'rb') as f:\n",
+        "                letter_set = pickle.load(f)\n",
+        "                if valid_dataset is not None:\n",
+        "                    valid_dataset[start_v:end_v, :, :] = letter_set[:vsize_per_class, :, :]\n",
+        "                    valid_labels[start_v:end_v] = label\n",
+        "                    start_v += vsize_per_class\n",
+        "                    end_v += vsize_per_class\n",
+        "                train_dataset[start_t:end_t, :, :] = letter_set[vsize_per_class:vsize_per_class+tsize_per_class, :, :]\n",
+        "                train_labels[start_t:end_t] = label\n",
+        "                start_t += tsize_per_class\n",
+        "                end_t += tsize_per_class\n",
+        "        except Exception as e:\n",
+        "          print('Unable to process data from', pickle_file, ':', e)\n",
+        "          raise\n",
+        "    \n",
+        "    return valid_dataset, valid_labels, train_dataset, train_labels\n",
+        "            \n",
+        "            \n",
         "train_size = 200000\n",
         "valid_size = 10000\n",
+        "test_size = 10000\n",
+        "\n",
+        "valid_dataset, valid_labels, train_dataset, train_labels = merge_datasets(train_datasets, train_size, valid_size)\n",
+        "__, __, test_dataset, test_labels = merge_datasets(test_datasets, test_size)\n",
         "\n",
-        "valid_dataset = train_dataset[:valid_size,:,:]\n",
-        "valid_labels = train_labels[:valid_size]\n",
-        "train_dataset = train_dataset[valid_size:valid_size+train_size,:,:]\n",
-        "train_labels = train_labels[valid_size:valid_size+train_size]\n",
-        "print('Training', train_dataset.shape, train_labels.shape)\n",
-        "print('Validation', valid_dataset.shape, valid_labels.shape)"
+        "print('Training:', train_dataset.shape, train_labels.shape)\n",
+        "print('Validation:', valid_dataset.shape, valid_labels.shape)\n",
+        "print('Testing:', test_dataset.shape, test_labels.shape)"
       ],
       "outputs": [
         {
           "output_type": "stream",
           "text": [
             "Training (200000, 28, 28) (200000,)\n",
```

Comment:
Have you verified that subsequent assignments run well and give sensible results with this default training set size?